src/main/java/bio/terra/cli/businessobject/resource/DataSource.java

```diff
@@ -70,11 +71,25 @@ protected void deleteControlled() {
   }
 
   /** Resolve data source. */
-  // TODO(PF-1743, PF-1744): Implement
   public String resolve() {
     return "";
   }
 
+  public String resolve(String resourceName) {
+    var resources = getDataSourceWorkspace().getResources();
+    return resources.stream()
+        .filter(
+            resource ->
+                resource.getResourceType() != Type.DATA_SOURCE
```

Comment:
Do we allow nested data sources and not display them, or are they just disallowed?